### PR TITLE
Allow GC to discard content after successful pull and unpack

### PR DESCRIPTION
**Pull request by ktock**
Related: https://github.com/containerd/containerd/issues/4304 https://github.com/containerd/containerd/pull/4414

This commit adds a config flag for allowing GC to clean layer contents up after unpacking these contents completed, which leads to deduplication of layer contents between the snapshotter and the content store.
This PR includes vendoring the latest containerd which includes the commit https://github.com/containerd/containerd/commit/d184a0a3430dc4a17a47cce37fb36126ac0c699a . (full diff: https://github.com/containerd/containerd/compare/v1.4.0-beta.0...d184a0a3430dc4a17a47cce37fb36126ac0c699a )

One thing I'm considering is the default value of this flag. In this patch, this "DiscardContent mode" is disabled by default but please tell me if it should be enabled by default.

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -80,6 +80,11 @@ type ContainerdConfig struct {
 	// related information) to snapshotters. These annotations are required by
 	// stargz snapshotter (https://github.com/containerd/stargz-snapshotter).
 	DisableSnapshotAnnotations bool `toml:"disable_snapshot_annotations" json:"disableSnapshotAnnotations"`
+
+	// DiscardContent is a boolean flag to specify whether to allow GC to clean
+	// layers up from the content store after successfully unpacking these
+	// contents to the snapshotter.
```
Comment (mikebrow) [inline pkg/config/config.go:86]: please add text for discard_content (and disable_snapshot_annotations while at it :-) to https://github.com/containerd/cri/blame/master/docs/config.md#L108

File: pkg/config/config.go [hunk 2]
```diff
@@ -80,6 +80,11 @@ type ContainerdConfig struct {
 	// related information) to snapshotters. These annotations are required by
 	// stargz snapshotter (https://github.com/containerd/stargz-snapshotter).
 	DisableSnapshotAnnotations bool `toml:"disable_snapshot_annotations" json:"disableSnapshotAnnotations"`
+
+	// DiscardContent is a boolean flag to specify whether to allow GC to clean
```
Comment (mikebrow) [inline pkg/config/config.go:84]: suggest DiscardUnpackedLayers or DiscardUnpackedContainerImageLayers

Comment (mikebrow) [reply pkg/config/config.go:84]: s/to clean/to remove/

File: pkg/config/config.go (hunk 1 above)
Comment (mikebrow) [reply pkg/config/config.go:86]: please indicate the default is false, layers are not removed from the content store during garbage collection.

Comment (mikebrow) [inline pkg/config/config.go:86]: s/contents/layers/

**Comment by ktock** [comment]
@mikebrow Thanks for comments. Fixed the patch based on the comments. PTAL.

**Pull request merged**